automerge-backend/src/backend.rs

```diff
@@ -1,22 +1,27 @@
 use core::cmp::max;
-use std::collections::{HashMap, HashSet, VecDeque};
+use std::{
+    collections::{HashMap, HashSet, VecDeque},
+    fmt::Debug,
+};
 
 use amp::ChangeHash;
 use automerge_protocol as amp;
 
 use crate::{
-    actor_map::ActorMap, change::encode_document, error::AutomergeError, internal::ObjectId,
-    op_handle::OpHandle, op_set::OpSet, pending_diff::PendingDiff, Change,
+    actor_map::ActorMap, change::encode_document, error::AutomergeError,
+    event_handlers::EventHandlers, internal::ObjectId, op_handle::OpHandle, op_set::OpSet,
+    pending_diff::PendingDiff, Change,
 };
 
-#[derive(Debug, PartialEq, Clone)]
+#[derive(Debug, Clone)]
 pub struct Backend {
     queue: Vec<Change>,
     op_set: OpSet,
     states: HashMap<amp::ActorId, Vec<usize>>,
     actors: ActorMap,
     history: Vec<Change>,
     history_index: HashMap<amp::ChangeHash, usize>,
+    pub event_handlers: EventHandlers,
```

Comment:
Do we want to make this public rather than proxying the add and remove handler methods? Feels like it's coupling us to this particular implementation strategy unnecessarily.

Reply:
Yeah, I think I'll change this to proxy the functions to give us the adaptability.